### PR TITLE
Add option to not generate default configuration in eos_cli_config_gen

### LINKED ISSUE #814: Provide capability to leverage eos_cli_config_gen to generate simple configlets
**Issue Type**

- [x] Template enhancement
- [ ] Role enhancement
- [ ] Documentation enhancement (avd.sh)

Currently, when leveraging eos_cli_config_gen, some default data is generated, which prevents you from using the role in the case of generating simple configlets.  There should be knobs to enable/disable this data from being generated.

```eos
!RANCID-CONTENT-TYPE: arista
!
transceiver qsfp default-mode 4x10G
!
hostname {{ inventory_hostname }}
```

Additionally, knobs should be provided to:
- Set the intended configuration path.
- Set documentation generation path.
- Knob to enable/disable documentation generation.

### PR DESCRIPTION
**Pull request by ClausHolbechArista**
## Change Summary

Add option to not generate default configuration in eos_cli_config_gen

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes)
- [ ] Documentation content changes
- [ ] Other (please describe):

## Related Issue(s)

Part of fix for #814 

## Component(s) name

`arista.avd.eos_cli_config_gen`

## Proposed changes

New knob for `eos_cli_config_gen`:
```yaml
generate_default_config: < true | false | default -> true >
```

## How to test

Added molecule scenario. No other configs are affected.

## Checklist:

- [x] My code has been rebased from devel before I start
- [x] I have read the [**CONTRIBUTING**](https://avd.sh/en/latest/docs/contributing/) document.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have updated [molecule CI](https://github.com/aristanetworks/ansible-avd/tree/devel/ansible_collections/arista/avd/molecule) testing accordingly
- [x] All new and existing tests passed ([`pre-commit`](https://www.avd.sh/docs/installation/development/#python-virtual-environment), `make linting` and `make sanity-lint`).

### CODE REVIEW

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/end.j2 [hunk 1]
```diff
@@ -0,0 +1,5 @@
+{# hostname #}
```
Comment (onurgashi) [inline ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/end.j2]: Should this be end?

**Review by onurgashi — APPROVED** [review]

File: ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/end.j2 (hunk 1 above)
Comment (ClausHolbechArista) [reply ansible_collections/arista/avd/roles/eos_cli_config_gen/templates/eos/end.j2]: haha no. Thank you for spotting this. Too much copy/paste.

**Comment by ClausHolbechArista** [comment]
@carlbuchmann please take a look at this one.

**Review by carlbuchmann — APPROVED** [review]
LGTM!

**Review by ClausHolbechArista — COMMENTED** [review]
LGTM

**Pull request merged**